crates/ide-assists/src/handlers/generate_function.rs

```diff
@@ -399,6 +394,26 @@ fn get_method_target(
     Some((target.clone(), get_insert_offset(&target)))
 }
 
+fn static_method_target(
```

Comment:
This was the simplest way to achieve the goal here, but I don't love the result.

Using mutable out-parameters like this doesn't feel very idiomatic, but it definitely works.
Removing the out parameters requires significantly more restructuring, and will be quite a bit more work to review.

Reply:
I'm happy to make a bigger change that removes these out params, but I didn't want to overwhelm reviewers with a bigger change than necessary. Please let me know if y'all think I should go further in this PR, or clean things up separately in a follow on change.

Reply:
I don't think doing that should change too much, and if you split the commits properly, that is you don't touch this commit again that shouldn't be any extra work to review

Reply:
Okay, there are a series of stepwise commits, mostly using assists, that get rid of the out params. I think there are further opportunities to reduce duplication in this assist, but that might be out of scope for now.